nifi-nar-bundles/nifi-ocr-bundle/nifi-ocr-processors/src/main/java/org/apache/nifi/processors/ocr/TesseractOCRProcessor.java

```diff
@@ -0,0 +1,359 @@
+import org.apache.nifi.components.AllowableValue;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.ValidationContext;
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.io.StreamCallback;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import net.sourceforge.tess4j.ITesseract;
+import net.sourceforge.tess4j.Tesseract;
+import net.sourceforge.tess4j.TesseractException;
+
+@Tags({"ocr", "tesseract", "image", "text"})
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@CapabilityDescription("Extracts text from images using Optical Character Recognition (OCR). The images are pulled from the incoming" +
+        " Flowfile's content. Supported image types are TIFF, JPEG, GIF, PNG, BMP, and PDF. Any Flowfile that doesn't contain" +
+        " a supported image type in its content body will be routed to the 'unsupported image format' relationship and no OCR will be performed." +
+        " This processor uses Tesseract to perform its duties and part of that requires that a valid Tesseract data (Tessdata) directory" +
+        " be specified in the 'Tessdata Directory' Property. This processor considers a valid Tessdata directory to be an existing directory on the" +
+        " local NiFi instance that contains one or more files ending with the '.traineddata' extension. The list of supported languages" +
+        " is built from the Tessdata directory configured by listing all files ending with '.traineddata' and considering those" +
+        " Tesseract language models. You can create you own Tesseract language models and place them in your Tessedata directory" +
+        " and the processor will display it in the dropdown list of languages available. All valid Tesseract configuration values" +
+        " may be passed to this processor by use of the 'Tesseract configuration values' which accepts a comma separated list" +
+        " of key=value pairs representing Tesseract configurations. 'Tesseract configuration values' is where all of your tuning" +
+        " values can be passed in to help increase the accuracy of your OCR operations based on your expected input images." +
+        " TesseractOCRProcessor only supports installations of Tesseract version 3.0 and greater.")
+public class TesseractOCRProcessor extends AbstractProcessor {
+
+    public static Set<String> SUPPORTED_LANGUAGES;
+    private static final String TESS_LANG_EXTENSION = ".traineddata";
+    private static List<AllowableValue> PAGE_SEGMENTATION_MODES;
+    private static volatile ITesseract tesseract;
```

Comment:
property descriptors on a given processor class should generally be static as they are immutable and do not hold the selected/chosen/entered value of a given property.  You are right that the property value is tied to an instance though that is covered already by how the API works so should be fair game to make the property descriptors themselves static.